### PR TITLE
fix(command-dev): newly added functions are not recognized

### PR DESCRIPTION
**Pull request by erezrokah**
**- Summary**

Fixes https://github.com/netlify/cli/issues/413

**- Test plan**

Run `netlify dev` and add a new function

**- A picture of a cute animal (not mandatory but encouraged)**
image

### CODE REVIEW

File: src/utils/serve-functions.js [hunk 1]
```diff
@@ -194,7 +204,19 @@ const createHandler = async function ({ dir, capabilities, omitFileChangesLog, w
...
   })
-  watcher.on('change', debouncedOnChange).on('unlink', debouncedOnUnlink)
+  watcher
+    .on('change', debouncedOnChange)
+    .on('unlink', debouncedOnUnlink)
+    .on('add', async (path) => {
+      logBeforeAction({ omitLog: omitFileChangesLog, path, action: 'added' })
+
+      await watcher.unwatch(watchDirs)
+      ;({ functions, watchDirs } = await getFunctionsAndWatchDirs(dir))
```
Comment (erezrokah) [inline src/utils/serve-functions.js:214]: `functions` is used a few lines down when handling requests when searching for a matching function

File: src/utils/serve-functions.js [hunk 2]
```diff
@@ -194,7 +204,19 @@ const createHandler = async function ({ dir, capabilities, omitFileChangesLog, w
     leading: false,
     trailing: true,
   })
-  watcher.on('change', debouncedOnChange).on('unlink', debouncedOnUnlink)
+  watcher
+    .on('change', debouncedOnChange)
+    .on('unlink', debouncedOnUnlink)
+    .on('add', async (path) => {
+      logBeforeAction({ omitLog: omitFileChangesLog, path, action: 'added' })
```
Comment (erezrokah) [inline src/utils/serve-functions.js:211]: This is not debounced since we want to handle all events

File: src/utils/serve-functions.js [hunk 3]
```diff
@@ -182,9 +192,9 @@ const validateFunctions = function ({ functions, capabilities, warn }) {
 const DEBOUNCE_WAIT = 300
 
 const createHandler = async function ({ dir, capabilities, omitFileChangesLog, warn }) {
-  const { functions, watchDirs } = await getFunctionsAndWatchDirs(dir)
+  let { functions, watchDirs } = await getFunctionsAndWatchDirs(dir)
   validateFunctions({ functions, capabilities, warn })
-  const watcher = chokidar.watch(watchDirs, { ignored: /node_modules/ })
+  const watcher = chokidar.watch(watchDirs, { ignored: /node_modules/, ignoreInitial: true })
```
Comment (erezrokah) [inline src/utils/serve-functions.js:197]: `ignoreInitial: true` otherwise `chokidar` emits `add` for all existing files

**Review by ehmicky — APPROVED** [review]

**Comment by ehmicky** [comment]
Note: CI tests are failing on Windows (I am not sure whether this is related)

**Comment by erezrokah** [comment]
> Note: CI tests are failing on Windows (I am not sure whether this is related)

Our Windows CI is flaky - trying to fix it in https://github.com/netlify/cli/pull/2188

**Pull request merged**